tests/sentry/rules/conditions/test_event_attribute.py

```diff
@@ -856,3 +856,15 @@ def test_attr_is_in(self):
             data={"match": MatchType.IS_IN, "attribute": "platform", "value": "python"}
         )
         self.assertDoesNotPass(rule, event)
+
+    def test_attr_not_in(self):
```

Comment:
i'd make sure to have a positive and negative test case for these.

Reply:
the tests include both the positive and negative test already

Reply:
in that case, it might be nice to split them out as separate test cases then to illustrate how this code works in a positive and negative case. it's really meant to help people understand the code and what expected / unexpected cases are handled.

---

in general my theory with this is that if we can improve our tests and readability of code - we can reduce the number of issues and magnitude of issues in the future. since we can't know when / where that will happen i think it's important we're diligent with our testing.